File: backend/blackbox/src/main/java/com/insight_web/blackbox/adapters/postgres/user/UserStorageImpl.java
package com.insight_web.blackbox.adapters.postgres.user;

import com.insight_web.blackbox.domain.User;
import com.insight_web.blackbox.domain.exception.ResourceNotFoundException;
import com.insight_web.blackbox.usecase.storage.UserStorage;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class UserStorageImpl implements UserStorage {
    private final UserRepository userRepository;
    @Override
    public User create(User user) {
        var entity = new UserEntity();
        entity.fillForSaving(user);
        return userRepository.save(entity).toDomainModel();
    }

    @Override
    public User findById(Long id) {
        return userRepository.findById(id).orElseThrow().toDomainModel();
    }

    @Override
    public User findByEmail(String email) {
        return userRepository
                .findByEmail(email)
                .orElseThrow(() -> new ResourceNotFoundException("user by email " + email + "not found"))
                .toDomainModel();
    }

    @Override
    public Boolean existsByEmail(String email) {
        return userRepository.existsByEmail(email);
    }
}

File: backend/auth/src/main/kotlin/com/insightweb/auth/adapters/postgres/user/UserStorageImpl.kt
package com.insightweb.auth.adapters.postgres.user

import com.insightweb.auth.domain.User
import com.insightweb.auth.domain.exception.ResourceNotFoundException
import com.insightweb.auth.usecase.storage.UserStorage
import org.springframework.stereotype.Repository
import kotlin.jvm.optionals.getOrNull

@Repository
class UserStorageImpl(
    private val userRepository: UserRepository,
): UserStorage {
    override fun create(user: User): User =
        userRepository.save(UserEntity().fillForSaving(user)).toDomainModel()

    override fun findById(id: Long): User = userRepository.findById(id).getOrNull()?.toDomainModel()
        ?: throw ResourceNotFoundException("user by $id not found")

    override fun findByEmail(email: String): User = userRepository.findByEmail(email)?.toDomainModel() ?: throw ResourceNotFoundException("user by email $email not found")

    override fun existsByEmail(email: String): Boolean = userRepository.existsByEmail(email)
}